e2e/mapimachines_test.go

```diff
@@ -0,0 +1,171 @@
+			ctx := context.Background()
+			machineList := &machinev1.MachineList{}
+			By(fmt.Sprintf("Listing worker machines in namespace: %s", machineAPINamespace))
+			workerLabelSelector := client.MatchingLabels{RoleLabel: "worker"}
+			err := cl.List(ctx, machineList, client.InNamespace(machineAPINamespace), workerLabelSelector)
+			Expect(err).NotTo(HaveOccurred(), "Failed to list worker machines")
+			Expect(machineList.Items).NotTo(BeEmpty(), "No worker machines found in the namespace to use as a template")
+
+			var templateMachine *machinev1.Machine
+			foundMapiMachine := false
+			for i, m := range machineList.Items {
+				if m.Spec.AuthoritativeAPI == "MachineAPI" {
+					templateMachine = &machineList.Items[i]
+					foundMapiMachine = true
+					break
+				}
+			}
+			if !foundMapiMachine {
+				Skip("No machine found with AuthoritativeAPI set to MachineAPI")
+			}
+			// Define the new machine based on the template.
+			newMachine = &machinev1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					// Using a predictable name based on the testcase in polarion.
+					Name:      fmt.Sprintf("%s-test-%s", templateMachine.Name, "ocp-81829"),
+					Namespace: templateMachine.Namespace,
+					Labels:    templateMachine.Labels,
+				},
+				Spec: *templateMachine.Spec.DeepCopy(),
+			}
+			// Clear status and other instance-specific fields that should not be copied.
+			newMachine.Spec.ProviderID = nil
+			newMachine.ObjectMeta.Labels = nil
+			By(fmt.Sprintf("Creating a new machine in namespace: %s", newMachine.Namespace))
+			// Create the new machine object in the cluster.
+			err = cl.Create(ctx, newMachine)
+			Expect(err).NotTo(HaveOccurred(), "Failed to create new machine")
+			By("Waiting for the new machine to enter 'Running' phase")
+			Eventually(komega.Object(newMachine), DefaultTimeout, DefaultInterval).
+				Should(HaveField("Status.Phase", HaveValue(Equal("Running"))),
+					fmt.Sprintf("Machine %s did not enter 'Running' phase within the timeout", newMachine.Name))
+
+			By("Verifying the new machine has a node reference")
+			Eventually(komega.Object(newMachine)).Should(HaveField("Status.NodeRef", Not(BeNil())))
+
+			By("Verifying status of mapi machine for synchronisation done successfully")
+			Expect(newMachine.Status.Conditions).To(ContainElement(
+				SatisfyAll(
+					HaveField("Type", ConditionSynchronized),
+					HaveField("Status", corev1.ConditionTrue),
+					HaveField("Message", ContainSubstring("Successfully synchronized MAPI Machine to CAPI")),
+				),
+			))
+
+			Expect(newMachine.Status.Conditions).To(ContainElement(
+				SatisfyAll(
+					HaveField("Type", MapiConditionPaused),
+					HaveField("Status", corev1.ConditionFalse),
+					HaveField("Message", ContainSubstring("The AuthoritativeAPI is set to MachineAPI")),
+				),
+			))
+
+			By("Verifying the corresponding machine in openshift-cluster-api is paused")
+			// Get all CAPI machines.
+			allMachines, err := framework.GetMachines(cl)
```

Comment:
Why not just get the machine directly? We know the name of it from the Machine we created? Why do we need to list and then filter?